server/controllers/message.controller.js

```diff
@@ -62,20 +69,82 @@ function send(req, res, next) {
         subject: req.body.subject,
         message: req.body.message,
         owner: req.body.from,
-        createdAt: createdTime,
-        readAt: createdTime,
-    });
-
+        created: new Date(),
+        readAt: new Date(),
+        isDeleted: false,
+    }).then(savedMessage => res.json(savedMessage))
+      .catch(e => next(e));
     // once the bulkCreate and create promises resolve, send the sender's saved message or an error
     Promise
         .join(bulkCreate, messageCreate, (bulkResult, messageResult) => res.json(messageResult))
         .catch(e => next(e));
 }
 
-function list() {}
+// returns a list of messages
+// Currently, there is no distinction between a message created by a user, &
+// a message sent by that user, since he is the 'owner' in both cases.
+// This needs integration with auth microservice
+// Query paramters: 'from', 'summary', 'limit'
+function list(req, res) {
+    const queryObject = {
+      where: {
+        isArchived: false
```

Comment:
@jsachs note I have already set the default to false here for lists. I have been able to find a way to add `archived:false` to the default scope and calling `unscope()` when querying for archived messages. In my experience however, queries that first unscope a previously scoped model are usually very expensive database operations and drastically increase the request time. Thoughts?

Reply:
I'm not sure I follow, though I don't know Sequelize scopes very well. Can we just have the default to be "no" for deleted and archived messages, and then add deleted and archived messages back in depending on the type of query?

Reply:
👍